### PR TITLE
PMM-12608 Add auto discovery limit for PG

### PR DESCRIPTION
**Pull request by matejkubinec**
**What this PR does / why we need it**:

Adds a new field for handling PG auto discovery limit. Missing is the text for the tooltip.

**Which issue(s) this PR fixes**:

PMM-12608

FB: https://github.com/Percona-Lab/pmm-submodules/pull/3424

**Special notes for your reviewer**:

image

### CODE REVIEW

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.messages.ts [hunk 1]
```diff
@@ -126,6 +130,7 @@ export const Messages = {
       postgresqlDetails: {
         database: 'Database name',
         maxQueryLength: 'Full Example (Fingerprint) storage is not allowed by default to have the best performance',
+        autoDiscoveryLimit: 'TODO',
```
Comment (matejkubinec) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.messages.ts:133]: Will be added once it's available.

Comment (matejkubinec) [reply public/app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.messages.ts]: Removed for now.

**Review by YashSartanpara1 — APPROVED** [review]

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/AdditionalOptions/AdditionalOptions.tsx [hunk 2]
```diff
@@ -41,16 +42,60 @@ export const AdditionalOptionsFormPart: FC<AdditionalOptionsFormPartProps> = ({
...
-    <h4>{Messages.form.labels.trackingOptions}</h4>
-    <RadioButtonGroupField
-      name="tracking"
-      data-testid="tracking-options-radio-button-group"
-      options={isRDS || isAzure ? rdsTrackingOptions : trackingOptions}
-    />
-  </>
-);
+export const PostgreSQLAdditionalOptions: FC<PostgreSQLAdditionalOptionsProps> = ({ form, isRDS, isAzure }) => {
+  const selectedOption = form.getState().values && form.getState().values.autoDiscoveryOptions;
```
Comment (doracretu3pillar) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/AdditionalOptions/AdditionalOptions.tsx:46]: an easier, much readable way 
const selectedOption = form.getState().values?.autoDiscoveryOptions ( returns 'undefined' or autoDiscoveryOptions )

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/AdditionalOptions/AdditionalOptions.tsx [hunk 3]
```diff
@@ -41,16 +42,60 @@ export const AdditionalOptionsFormPart: FC<AdditionalOptionsFormPartProps> = ({
...
+  const [selectedValue, setSelectedValue] = useState<string>(selectedOption || AutoDiscoveryOptionsInterface.disabled);
+  const autoDiscoveryValidators = useMemo(() => [validators.min(0)], []);
+  const styles = useStyles2(getStyles);
+
+  const getAutoDiscoveryLimitValue = (type: AutoDiscoveryOptionsInterface) => {
+    switch (type) {
```
Comment (doracretu3pillar) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/AdditionalOptions/AdditionalOptions.tsx:52]: `const getAutoDiscoveryLimitValue = (type: AutoDiscoveryOptionsInterface) =  type === AutoDiscoveryOptionsInterface.disabled ? 0 : 1000 `

**Review by doracretu3pillar — APPROVED** [review]

**Review by doracretu3pillar — APPROVED** [review]

**Pull request merged**